src/main/java/jenkins/branch/NoTriggerOrganizationFolderProperty.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.branch;
+
+import com.cloudbees.hudson.plugins.folder.AbstractFolderProperty;
+import com.cloudbees.hudson.plugins.folder.AbstractFolderPropertyDescriptor;
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Job;
+import hudson.model.Queue;
+import hudson.util.FormValidation;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Defines {@link NoTriggerBranchProperty} on selected branches.
+ */
+@Restricted(NoExternalUse.class)
+public class NoTriggerOrganizationFolderProperty extends AbstractFolderProperty<OrganizationFolder> {
+
+    /** Regexp of branch names to trigger. */
+    private final String branches;
+
+    @DataBoundConstructor
+    public NoTriggerOrganizationFolderProperty(String branches) {
+        this.branches = branches;
+    }
+    
+    public String getBranches() {
+        return branches;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractFolderPropertyDescriptor {
+
+        @Override
+        public String getDisplayName() {
+            return Messages.NoTriggerBranchProperty_suppress_automatic_scm_triggering();
+        }
+
+        public FormValidation doCheckBranches(@QueryParameter String value) {
+            try {
+                Pattern.compile(value);
+                return FormValidation.ok();
+            } catch (PatternSyntaxException x) {
+                return FormValidation.error(x.getMessage());
+            }
+        }
+
+    }
+
+    @Extension
+    public static class Dispatcher extends Queue.QueueDecisionHandler {
+
+        @SuppressWarnings({"unchecked", "rawtypes"}) // untypable
+        @Override
+        public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
+            for (Action action :  actions) {
+                if (action instanceof CauseAction) {
+                    for (Cause c : ((CauseAction) action).getCauses()) {
+                        if (c instanceof BranchIndexingCause) {
+                            if (p instanceof Job) {
+                                Job j = (Job) p;
+                                if (j.getParent() instanceof MultiBranchProject) {
```

Comment:
That would be nice! Except for the fact that you cannot generify a SAM interface based on exception list (`throws` clause), you could have

```java
public static <T,R> as(Object arg, Class<T> type, R fallback, Function<T,R> function) {
    if (type.isInstance(arg)) {
        return function(type.cast(arg));
    } else {
        return falback;
    }
}
```

used as

```java
if (as(j.getParent(), MultiBranchProject.class, false, (mbp) -> mbp.matchesSomeCondition())) {
    return false;
}
```